TabulateSmarterTestAdminPackage.Common/Validators/ValidatorCollection.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace TabulateSmarterTestAdminPackage.Common.Validators
+{
+    public class ValidatorCollection
+    {
+        private readonly Queue<Validator> _validators;
+
+        public ValidatorCollection()
+        {
+            _validators = new Queue<Validator>();
+        }
+
+        public ValidatorCollection(Validator validator)
+        {
+            _validators = new Queue<Validator>();
+            _validators.Enqueue(validator);
+        }
+
+        public ValidatorCollection(IEnumerable<Validator> validators)
+        {
+            _validators = new Queue<Validator>();
+            Add(validators);
+        }
+
+        public ValidatorCollection Add(Validator validator)
+        {
+            _validators.Enqueue(validator);
+            return this;
+        }
+
+        public ValidatorCollection Add(IEnumerable<Validator> validators)
+        {
+            foreach (var validator in validators)
+            {
+                _validators.Enqueue(validator);
+            }
+            return this;
+        }
+
+        public bool ObjectPassesValidation(object value)
+        {
+            return _validators.All(x => x.IsValid(value));
+        }
+
+        public string ObjectValidationErrors(object value)
+        {
+            return _validators
+                .Where(x => !x.IsValid(value))
+                .Select(x => x.GetMessage())
+                .Aggregate((i, j) => i + j);
```

Comment:
I was going to throw all of the different errors in brackets "[Error1][Error2] and return them all as a single row per item that's failing validation. I don't think there's any harm in returning all the things wrong with a particular field. The format would look like: $"{path} attribute {attribute} violates {violationMessage}" (AdminPackageUtility class)